compiler/rustc_arena/src/tests.rs

```diff
@@ -101,37 +101,6 @@ pub fn test_typed_arena_zero_sized() {
     }
 }
 
-#[test]
-pub fn test_typed_arena_clear() {
```

Comment:
Why do you remove these?

Reply:
I deleted `clear()`, which is unused in the compiler, so these no longer compile.

Reply:
There is plenty of public API code in the compiler that's only being used by tests. I don't think those tests should be removed. They still test/benchmark code that *is* used by the compiler.

Maybe this particular test can be removed, but maybe change the benches below to drop the arena instead? I don't think it's a good idea to remove tests for something because they use special code only used by tests for it.

Reply:
I can make sure a test for drop remains, but otherwise benchmarking dead code just makes the benchmark take longer IMO.